src/hammer-vlsi/hammer_vlsi.py

```diff
@@ -583,31 +642,46 @@ def check_input_files(self, extensions: List[str]) -> bool:
         return not error
 
     # TODO: should some of these live in hammer_tech instead?
-    def filter_and_select_libs(self, func: Callable[[hammer_tech.Library], List[str]],
+    def filter_and_select_libs(self,
+                               lib_filters: List[Callable[[hammer_tech.Library], bool]] = [],
+                               sort_func: Optional[Callable[[hammer_tech.Library], Union[Number, str, tuple]]] = None,
+                               extraction_func: Callable[[hammer_tech.Library], List[str]] = None,
                                extra_funcs: List[Callable[[str], str]] = [],
-                               lib_filters: List[Callable[[hammer_tech.Library], bool]] = []) -> List[str]:
+                               ) -> List[str]:
         """
         Generate a list by filtering the list of libraries and selecting some parts of it.
 
-        :param func: Function to call to extract the desired component of the lib.
-        :param extra_funcs: List of extra functions to call before wrapping them in the arg prefixes.
         :param lib_filters: Filters to filter the list of libraries before selecting desired results from them.
-        :return: List of arguments to pass to a shell script
+        e.g. remove libraries of the wrong type
+        :param sort_func: Sort function to re-order the resultant components.
+        e.g. put stdcell libraries before any other libraries
+        :param extraction_func: Function to call to extract the desired component of the lib.
+        e.g. turns the library into the ".lib" file corresponding to that library
+        :param extra_funcs: List of extra functions to call before wrapping them in the arg prefixes.
+
+        :return: List generated from list of libraries
         """
+        if extraction_func is None:
+            raise TypeError("extraction_func is required")
+
         filtered_libs = reduce_named(
             sequence=lib_filters,
             initial=self.technology.config.libraries,
             function=lambda libs, func: filter(func, libs)
         )
 
-        lib_results = list(reduce(lambda a, b: a+b, list(map(func, filtered_libs)))) # type: List[str]
+        if sort_func is not None:
+            filtered_libs = sorted(filtered_libs, key=sort_func)
```

Comment:
I see. Python's `sorted()` function assigns each element a sort value using `key`, so it doesn't appear to be possible to drop values.

If you mean why we need to check if sort_func is not None, that's because Python doesn't know how to compare HammerLibrary objects by default.